src/main/java/org/openlmis/requisition/domain/requisition/RequisitionLineItem.java

```diff
@@ -391,6 +391,10 @@ void updateFrom(RequisitionLineItem requisitionLineItem) {
     if (requisition.isApprovable()) {
       this.approvedQuantity = requisitionLineItem.getApprovedQuantity();
       this.remarks = requisitionLineItem.getRemarks();
+      RequisitionTemplate requisitionTemplate = requisition.getTemplate();
+      if (requisitionTemplate != null && requisitionTemplate.isPatientsTabEnabled()) {
+        this.totalReceivedQuantity = requisitionLineItem.getTotalReceivedQuantity();
+      }
```

Comment:
Looks good.
Consider adding tests like [this one](https://github.com/OpenLMIS/openlmis-requisition/blob/master/src/test/java/org/openlmis/requisition/domain/requisition/RequisitionLineItemTest.java#L165) to cover this case.

Reply:
added